cmd/describe/breakglasscredential/cmd.go

```diff
@@ -0,0 +1,129 @@
+	Short:   "Show details of a break glass credential on a cluster",
+	Long:    "Show details of a break glass credential on a cluster.",
+	Example: `  # Show details of a break glass credential with ID "12345" on a cluster named "mycluster"
+  rosa describe break-glass-credential 12345 --cluster=mycluster `,
+	Run:    run,
+	Hidden: true,
+	Args:   cobra.MaximumNArgs(1),
+}
+
+var args struct {
+	id string
+}
+
+func init() {
+	flags := Cmd.Flags()
+	flags.SortFlags = false
+	ocm.AddClusterFlag(Cmd)
+	output.AddFlag(Cmd)
+	flags.StringVar(
+		&args.id,
+		"id",
+		"",
+		"Id for the break glass credential of the cluster to target",
+	)
+}
+
+func run(cmd *cobra.Command, argv []string) {
+	r := rosa.NewRuntime().WithAWS().WithOCM()
+	defer r.Cleanup()
+	err := runWithRuntime(r, cmd, argv)
+	if err != nil {
+		r.Reporter.Errorf(err.Error())
+		os.Exit(1)
+	}
+}
+
+func runWithRuntime(r *rosa.Runtime, cmd *cobra.Command, argv []string) error {
+	breakGlassCredentialId := args.id
+	// Allow the use also directly set the break glass credential id as positional parameter
+	if len(argv) == 1 && !cmd.Flag("id").Changed {
+		breakGlassCredentialId = argv[0]
+	}
+	if breakGlassCredentialId == "" {
+		return fmt.Errorf("you need to specify a break glass credential id with '--id' parameter")
+	}
+	clusterKey := r.GetClusterKey()
+	cluster := r.FetchCluster()
+	if cluster.State() != cmv1.ClusterStateReady {
+		return r.Reporter.Errorf("Cluster '%s' is not yet ready", clusterKey)
+	}
+
+	if !cluster.Hypershift().Enabled() {
+		return fmt.Errorf(
+			"break glass credential is only supported for Hosted Control Planes",
+		)
+	}
+
+	r.Reporter.Debugf("Fetching the break glass credential '%s' for cluster '%s'", breakGlassCredentialId, clusterKey)
+
+	breakGlassCredentialConfig, err := r.OCMClient.GetBreakGlassCredential(cluster.ID(), breakGlassCredentialId)
+	if err != nil {
+		return err
+	}
+
+	output := describeBreakGlassCredential(r, cluster, clusterKey, breakGlassCredentialConfig)
+
+	fmt.Print(output)
+	return nil
+}
+
+func describeBreakGlassCredential(r *rosa.Runtime, cluster *cmv1.Cluster,
+	clusterKey string, config *cmv1.BreakGlassCredential) string {
+	breakGlassCredentialOutput := fmt.Sprintf("\n"+
+		"ID:                                    %s\n"+
+		"Cluster ID:                            %s\n"+
+		"Username:                              %s\n"+
+		"Expire at:                             %s\n"+
+		"Status:                                %s\n",
+		config.ID(),
+		cluster.ID(),
+		config.Username(),
+		config.ExpirationTimestamp().Format("Jan _2 2006 15:04:05 MST"),
+		config.Status(),
+	)
+
+	revocationTimeStamp, ok := config.GetRevocationTimestamp()
+	if ok {
+		breakGlassCredentialOutput = fmt.Sprintf("%s"+
+			"Revoked at:                            %s\n",
+			breakGlassCredentialOutput,
+			revocationTimeStamp.Format("Jan _2 2006 15:04:05 MST"),
+		)
+	}
```

Comment:
missing the kubeconfig if present

Reply:
Discussed with @andreadecorte , added a message to extract kubeconfig